lib/modules/platform/gerrit/index.ts

```diff
@@ -0,0 +1,553 @@
+  getGerritRepoUrl,
+  mapBranchStatusToLabel,
+  mapGerritChangeToPr,
+  mapPrStateToGerritFilter,
+} from './utils';
+
+export const id = 'gerrit';
+
+const defaults: {
+  endpoint?: string;
+  hostType: string;
+} = {
+  hostType: 'gerrit',
+};
+
+let config: {
+  repository?: string;
+  head?: string;
+  config?: GerritProjectInfo;
+  labels: { [key: string]: GerritLabelTypeInfo };
+  labelMappings?: {
+    stabilityDaysLabel?: string;
+    mergeConfidenceLabel?: string;
+  };
+  gerritUsername?: string;
+} = {
+  labels: {},
+  labelMappings: {
+    stabilityDaysLabel: 'Renovate-Stability',
+  },
+};
+
+export function mergeToConfig(newConfig: typeof config): void {
+  config = { ...config, ...newConfig };
+}
+
+export function initPlatform({
+  endpoint,
+  username,
+  password,
+  gerritLabelMapping,
+}: PlatformParams & RepoGlobalConfig): Promise<PlatformResult> {
+  logger.info(`initPlatform(${endpoint!}, ${username!})`);
+  if (!endpoint) {
+    throw new Error('Init: You must configure a Gerrit Server endpoint');
+  }
+  if (!(username && password)) {
+    throw new Error(
+      'Init: You must configure a Gerrit Server username/password'
+    );
+  }
+  config.labelMappings = gerritLabelMapping;
+  config.gerritUsername = username;
+  defaults.endpoint = ensureTrailingSlash(endpoint);
+  setBaseUrl(defaults.endpoint);
+  const platformConfig: PlatformResult = {
+    endpoint: defaults.endpoint,
+  };
+  return Promise.resolve(platformConfig);
+}
+
+/**
+ * Get all state="ACTIVE" and type="CODE" repositories from gerrit
+ */
+export async function getRepos(): Promise<string[]> {
+  logger.debug(`getRepos()`);
+  return await client.getRepos();
+}
+
+/**
+ * Clone repository to local directory
+ * @param config
+ */
+export async function initRepo({
+  repository,
+  endpoint,
+  gitUrl,
+}: RepoParams): Promise<RepoResult> {
+  logger.info(`initRepo(${repository}, ${endpoint!}, ${gitUrl!})`);
+  const projectInfo = await client.getProjectInfo(repository);
+  const branchInfo = await client.getBranchInfo(repository);
+
+  config = {
+    ...config,
+    repository,
+    head: branchInfo.revision,
+    config: projectInfo,
+    labels: projectInfo.labels ?? {},
+  };
+  const baseUrl = endpoint ?? defaults.endpoint!;
+  const url = getGerritRepoUrl(repository, baseUrl);
+  configureScm(repository, config.gerritUsername!);
+
+  // Initialize Git storage
+  await git.initRepo({ url });
+  await git.syncGit(); //if not called the hook can be removed later...
+
+  //abandon "open" and "rejected" changes at startup
+  const rejectedChanges = await findOwnPr({
+    branchName: '',
+    state: 'open',
+    label: '-2',
+  });
+  for (const change of rejectedChanges) {
+    await client.abandonChange(change._number);
+  }
+  const repoConfig: RepoResult = {
+    defaultBranch: config.head!,
+    isFork: false,
+    repoFingerprint: repoFingerprint(repository, baseUrl), //TODO: understand the semantic? what cache could be stale/wrong?
```

Comment:
on github repos can be moved or recreated. So this is used to detect that this repo is new and the cache should be invalidated. Does gerrit has a unique id for a repo? if yes, use that.

Reply:
okay, it's for [this](https://docs.renovatebot.com/self-hosted-configuration/#repositorycache) (default: disabled) feature?

I don't think there exists [a unique id](https://gerrit-review.googlesource.com/Documentation/rest-api-projects.html#project-info) for this use-case. 
You can delete a project and recreate it with same name/description but completely different content...

But i think that's not a real-world scenario and using the baseUrl+repository-name is good enough.